client.go

```diff
@@ -0,0 +1,101 @@
+package youtube
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+// Client offers methods to download video metadata and video streams.
+type Client struct {
+	// Debug enables debugging output through log package
+	Debug bool
+
+	// HTTPClient can be used to set a custom HTTP client.
+	// If not set, http.DefaultClient will be used
+	HTTPClient *http.Client
+}
+
+// GetVideo fetches video metadata
+func (c *Client) GetVideo(url string) (*Video, error) {
+	return c.GetVideoContext(context.Background(), url)
+}
+
+// GetVideoContext fetches video metadata with a context
+func (c *Client) GetVideoContext(ctx context.Context, url string) (*Video, error) {
+	id, err := extractVideoID(url)
+	if err != nil {
+		return nil, fmt.Errorf("extractVideoID failed: %w", err)
+	}
+
+	eurl := "https://youtube.googleapis.com/v/" + id
+	resp, err := c.httpGet(ctx, "https://youtube.com/get_video_info?video_id="+id+"&eurl="+eurl)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	resp.Body.Close()
```

Comment:
It seems to me that close should be  after error check with defer.

Reply:
done